### PR TITLE
Hyphenated standard neutron names

### LINKED ISSUE #1042: Convention-breaking pulse-time and other labels
Mantid converters should use `'pulse-time'` and not `'pulse_times'`. The latter breaks two conventions (underscore instead of hyphen, plural where other labels are singular). The same issues exists for others such `source-position` and `sample-position`.

- Fix converters.
- Fix `neutron::convert` or rather the things in `neutron/beamline.cpp`
- Fix documentation, if applicable.
- Check if there are others I did not mention above.

### PR DESCRIPTION
**Pull request by OwenArnold**
Fixes #1042

### CODE REVIEW

File: python/neutron.cpp [hunk 1]
```diff
@@ -21,13 +21,13 @@ template <class T> void bind_positions(py::module &m) {
     :return: A variable containing the detector pixel positions.
     :rtype: Variable)");
 
-  m.def("source_position", py::overload_cast<T>(source_position), R"(
+  m.def("source-position", py::overload_cast<T>(source_position), R"(
```
Comment (SimonHeybrock) [inline python/neutron.cpp:24]:
```suggestion
  m.def("source_position", py::overload_cast<T>(source_position), R"(
```
Can't have hyphen in function name

File: python/neutron.cpp [hunk 2]
```diff
@@ -21,13 +21,13 @@ template <class T> void bind_positions(py::module &m) {
...
     Extract the neutron source position from a data array or a dataset.
 
     :return: A scalar variable containing the source position.
     :rtype: Variable)");
 
-  m.def("sample_position", py::overload_cast<T>(sample_position), R"(
+  m.def("sample-position", py::overload_cast<T>(sample_position), R"(
```
Comment (SimonHeybrock) [inline python/neutron.cpp:30]:
```suggestion
  m.def("sample_position", py::overload_cast<T>(sample_position), R"(
```

**Comment by OwenArnold** [comment]
> No changes in docs required?

No.

**Comment by SimonHeybrock** [comment]
@OwenArnold CI is failing.

**Comment by OwenArnold** [comment]
> @OwenArnold CI is failing.

Yes will address.

**Pull request merged**